src/coreclr/System.Private.CoreLib/src/System/Buffer.CoreCLR.cs

```diff
@@ -15,8 +16,9 @@ public partial class Buffer
         [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "Buffer_MemMove")]
         private static unsafe partial void __Memmove(byte* dest, byte* src, nuint len);
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern void __BulkMoveWithWriteBarrier(ref byte destination, ref byte source, nuint byteCount);
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "Buffer_BulkMoveWithWriteBarrier")]
+        [SuppressGCTransition]
+        private static partial void __BulkMoveWithWriteBarrier(ref byte destination, ref byte source, nuint byteCount);
```

Comment:
@AaronRobinsonMSFT @jkotas if we mark it as a QCall, I presume we can delete [this code](https://github.com/dotnet/runtime/blob/a95c2317d882fdabcec7ef3a22a4ebbe02672ed6/src/libraries/System.Private.CoreLib/src/System/Buffer.cs#L188-L219) that tries to be smart and avoid GC starvation for what previously was an FCall. 

(assuming the overhead from QCall is acceptable, I guess it might show up for small inputs)

Reply:
Although, I am not sure GC can interrupt during `memcpy`?

Reply:
I think we would still need this. The GC Poll is only on the return, not during the copy. I think we could still suffer from starvation if didn't break it up.

I didn't really change anything here other than make the existing GCPoll implicit instead of explicit.